File: src/types/clothes.ts
export type ClothingType = 
  | 'collared_tshirts'
  | 'tshirts_without_collar'
  | 'uniform_shirts'
  | 'uniforms'
  | 'uniform_pants'
  | 'bags'
  | 'aprons'
  | 'cloth_vests'
  | 'reflective_vests'
  | 'thick_caps'
  | 'simple_caps'
  | 'towels'
  | 'sheets'
  | 'aprons_kitchen'
  | 'other';

export type ServiceType = 'embroidery' | 'stamping' | 'dtf' | 'transfer';

export type ServiceLocation = 
  | 'front_right'
  | 'front_left'
  | 'back'
  | 'sleeve_left'
  | 'sleeve_right'
  | 'center_front'
  | 'center_back'
  | 'left_side'
  | 'right_side'
  | 'top'
  | 'bottom'
  | 'custom';

export type ClothingSize = 'S' | 'M' | 'L' | 'XL' | 'XXL' | 'XXXL';

export type SizesMap = Record<ClothingSize, number>;

export interface ClothingService {
  id: string;
  clothes_id: string;
  service_type: ServiceType;
  location: ServiceLocation;
  description?: string;
  unit_price: number;
  created_at: string;
  updated_at: string;
}

export interface Clothes {
  id: string;
  order_id: string;
  clothing_type: ClothingType;
  custom_type?: string;
  unit_price: number;
  sizes: SizesMap;
  color: string;
  total_quantity: number;
  services: ClothingService[];
  created_at: string;
  updated_at: string;
}

export interface CreateClothingService {
  service_type: ServiceType;
  location: ServiceLocation;
  description?: string;
  unit_price: number;
}

export interface CreateClothes {
  order_id: string;
  clothing_type: ClothingType;
  custom_type?: string;
  unit_price: number;
  sizes: SizesMap;
  color: string;
  services: CreateClothingService[];
}

export interface UpdateClothes {
  clothing_type?: ClothingType;
  custom_type?: string | null;
  unit_price?: number;
  sizes?: SizesMap;
  color?: string;
}

export interface UpdateClothingService {
  service_type?: ServiceType;
  location?: ServiceLocation;
  description?: string | null;
  unit_price?: number;
}

// Labels for UI
export const CLOTHING_TYPE_LABELS: Record<ClothingType, string> = {
  collared_tshirts: 'Camisetes de Gola',
  tshirts_without_collar: 'Camisetes sem Gola',
  uniform_shirts: 'Camisas de Uniformes',
  uniforms: 'Fardamentos',
  uniform_pants: 'Calças de Fardamentos',
  bags: 'Bolços',
  aprons: 'Batas',
  cloth_vests: 'Coletes de Pano',
  reflective_vests: 'Coletes Refletore',
  thick_caps: 'Bones Grossos',
  simple_caps: 'Bones Simples',
  towels: 'Toalhas',
  sheets: 'Lençois',
  aprons_kitchen: 'Aventais',
  other: 'Outros (especificar)'
};

export const SERVICE_TYPE_LABELS: Record<ServiceType, string> = {
  embroidery: 'Bordado',
  stamping: 'Estampagem',
  dtf: 'DTF',
  transfer: 'Transfer'
};

export const SERVICE_LOCATION_LABELS: Record<ServiceLocation, string> = {
  front_right: 'Frente Direita',
  front_left: 'Frente Esquerda',
  back: 'Atrás',
  sleeve_left: 'Manga Esquerda',
  sleeve_right: 'Manga Direita',
  center_front: 'Centro Frente',
  center_back: 'Centro Atrás',
  left_side: 'Lado Esquerdo',
  right_side: 'Lado Direito',
  top: 'Topo',
  bottom: 'Base',
  custom: 'Personalizado'
};

export const CLOTHING_SIZES: ClothingSize[] = ['S', 'M', 'L', 'XL', 'XXL', 'XXXL'];

File: src-tauri/src/models/clothes.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use time::OffsetDateTime;
use std::collections::HashMap;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum ClothingType {
    #[serde(rename = "collared_tshirts")]
    CollaredTshirts,
    #[serde(rename = "tshirts_without_collar")]
    TshirtsWithoutCollar,
    #[serde(rename = "uniform_shirts")]
    UniformShirts,
    #[serde(rename = "uniforms")]
    Uniforms,
    #[serde(rename = "uniform_pants")]
    UniformPants,
    #[serde(rename = "bags")]
    Bags,
    #[serde(rename = "aprons")]
    Aprons,
    #[serde(rename = "cloth_vests")]
    ClothVests,
    #[serde(rename = "reflective_vests")]
    ReflectiveVests,
    #[serde(rename = "thick_caps")]
    ThickCaps,
    #[serde(rename = "simple_caps")]
    SimpleCaps,
    #[serde(rename = "towels")]
    Towels,
    #[serde(rename = "sheets")]
    Sheets,
    #[serde(rename = "aprons_kitchen")]
    ApronsKitchen,
    #[serde(rename = "other")]
    Other,
}

impl Default for ClothingType {
    fn default() -> Self {
        ClothingType::CollaredTshirts
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum ServiceType {
    #[serde(rename = "embroidery")]
    Embroidery,
    #[serde(rename = "stamping")]
    Stamping,
    #[serde(rename = "dtf")]
    Dtf,
    #[serde(rename = "transfer")]
    Transfer,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum ServiceLocation {
    #[serde(rename = "front_right")]
    FrontRight,
    #[serde(rename = "front_left")]
    FrontLeft,
    #[serde(rename = "back")]
    Back,
    #[serde(rename = "sleeve_left")]
    SleeveLeft,
    #[serde(rename = "sleeve_right")]
    SleeveRight,
    #[serde(rename = "center_front")]
    CenterFront,
    #[serde(rename = "center_back")]
    CenterBack,
    #[serde(rename = "left_side")]
    LeftSide,
    #[serde(rename = "right_side")]
    RightSide,
    #[serde(rename = "top")]
    Top,
    #[serde(rename = "bottom")]
    Bottom,
    #[serde(rename = "custom")]
    Custom,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
pub enum ClothingSize {
    #[serde(rename = "S")]
    S,
    #[serde(rename = "M")]
    M,
    #[serde(rename = "L")]
    L,
    #[serde(rename = "XL")]
    XL,
    #[serde(rename = "XXL")]
    XXL,
    #[serde(rename = "XXXL")]
    XXXL,
}

// Type alias for sizes mapping
pub type SizesMap = HashMap<ClothingSize, u32>;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Clothes {
    pub id: String,
    pub order_id: String,
    pub clothing_type: String, // Will be converted to/from ClothingType in DTOs
    pub custom_type: Option<String>,
    pub unit_price: f64,
    pub sizes: String, // JSON string, will be parsed to/from SizesMap in DTOs
    pub color: String,
    pub total_quantity: i32,
    pub created_at: OffsetDateTime,
    pub updated_at: OffsetDateTime,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct ClothingService {
    pub id: String,
    pub clothes_id: String,
    pub service_type: String, // Will be converted to/from ServiceType in DTOs
    pub location: String, // Will be converted to/from ServiceLocation in DTOs
    pub description: Option<String>, // New description field
    pub unit_price: f64,
    pub created_at: OffsetDateTime,
    pub updated_at: OffsetDateTime,
}
